DOCUMENTATION.md

```diff
@@ -43,6 +45,190 @@ The overall territory is laid out here in this [C4](https://c4model.com) System
+   - Each physical Partition maps to a Tranche, so processing still honors stream concurrency limits, even if the CFP library assigns more than one partition to a given instance.
+   - The CFP library provides for estimation of the Lag of documents to be processed between the current checkpoint position and the tail of each partition, which can surfaced as a metric.
+   - Over time, as a Container splits into more physical partitions (either due to assigning > 10K RU per physical partition, or by exceeding the 50GB/partition limit), more Partitions will arise.
+   - Checkpoints are maintained by the CFP library within an auxiliary (by convention, a sister container with the name`{container}-aux`) CosmosDB container (the document containing the checkpoint doubles as a lease management target)
+2. `Propulsion.DynamoStore`: Reads from an Index Table maintained by a `Propulsion.DynamoStore.Indexer` Lambda
+    - In the current implementation, the Indexer feeds all items into a single Tranche within its index (extending it to support multiple Tranches based on a hash of the stream identifier would not be difficult).
+    - If an index was to present multiple tranches, the Reader is ready to handle them as is (although there's no support within the source for load balancing (splitting reads by Stream Name hash and then assigning shards across all active instances) across multiple processor instances).
+3. `Propulsion.EventStoreDb`: Uses the EventStoreDb gRPC based API to pull events from the `$all` stream.
+   - In the current implementation, there's no support within the source for load balancing (splitting reads by Stream Name hash and then assigning shards across all active instances) across multiple processor instances
+   - In the current implementation, there's no support for surfacing lag metrics on a continual basis (the reader does report it initially)
+   - There are facilities for storing checkpoints in CosmosStore, DynamoStore, Postgres, SQL Server. There is not presently a [checkpoint store implementation that maintains the checkpoints EventStoreDb itself at present](https://github.com/jet/propulsion/issues/8).
+4. `Propulsion.Feed`: Provides for reading from an arbitrary upstream system. Such a system might present an ATOM-like feed, but equally can be a periodic ingestion of a dataset from a pool of data that is not incrementally readable (such as a Table in a data warehouse). 
```

Comment:
```suggestion
4. `Propulsion.Feed`: Provides abstraction for reading events from an arbitrary upstream system. Such a system might present an ATOM-like feed. Equally, it can be a periodic ingestion of a dataset from a pool of data that is not incrementally readable (such as a Table in a data warehouse). 
```

`can be a periodic ingestion of a dataset from a pool of data that is not incrementally readable` - that'd be good to simplify. Is it about versioning?

Reply:
Hm. the readme.md explains it longer, I was trying to keep it short here.
You can either
1. expose something that can be checkpointed (but just via HTTP instead of a store change feed etc)
2. expose something that can't be checkpointed, and then there will be a checkpoint generated based on the time (and you tell it the interval you want to do a full walk/reingest)

Reply:
have added extended the description. It's stupid long now. But I guess its not abstract!